### PR TITLE
custom container and component exists validation

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: admin_command.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ type AdminCommand struct {
 	Alias     string       `yaml:"alias" json:"alias"`
 	Timeout   uint         `yaml:"timeout" json:"timeout"`
 	RunType   string       `yaml:"run_type" json:"run_type"`
-	Component string       `yaml:"component" json:"component"`
-	Image     CommandImage `yaml:"image" json:"image"`
+	Component string       `yaml:"component" json:"component" validate:"componentexists"`
+	Image     CommandImage `yaml:"image" json:"image"` // TODO: validate exists
```
Comment (emosbaugh) [inline admin_command.go:9]: this one is tough. i will have to write another validator. we really should try to collapse all these to use the same format.

File: validate.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		return fmt.Sprintf("[%d]", i) + rest, nil
+	}
+
+	return "", nil
+}
+
+func ComponentExists(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	root, ok := topStruct.Interface().(*RootConfig)
+	if !ok {
+		return true
```
Comment (divolgin) [inline validate.go:175]: why is this true?

File: validate.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	}
+
+	if field.Kind() != reflect.String {
+		return true
```
Comment (divolgin) [inline validate.go:180]: And this

File: validate.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+		}
+	}
+
+	return false
+}
+
+func ContainerExists(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	root, ok := topStruct.Interface().(*RootConfig)
+	if !ok {
+		return true
```
Comment (divolgin) [inline validate.go:202]: comment?

File: validate.go [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+			}
+			return false
+		}
+	}
+
+	// let componentexists show the error
+	return true
+}
+
+func ComponentContainerFormat(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
```
Comment (divolgin) [inline validate.go]: Why does this function return a boolean but is called `Format`?

**Comment by divolgin** [comment]
lgtm

**Comment by divolgin** [comment]
lgtm

**Pull request merged**